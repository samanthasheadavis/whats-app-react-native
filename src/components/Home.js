export default class Home extends React.Component {
  render(){
    return (
      <View style={styles.container }>
        <Button />
      </View>
    )
  }
} 
