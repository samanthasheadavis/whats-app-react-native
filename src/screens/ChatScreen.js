import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';<Button />

export default class Home extends React.Component {
  static navigationOptions = {
  title: "Chat Screen"
}
  render(){
    return (
      <Button />
    )
  }
}
