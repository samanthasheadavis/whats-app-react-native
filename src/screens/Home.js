import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

export default class Home extends React.Component {
  static navigationOptions = {
  title: "Home Screen"
}
  render(){
    return (
      <View style={styles.container }>
        <Button title='Navigate to Chat Screen' onPress={() => this.props.navigation.navigate('chat',{ name: 'Samantha'})}/>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
