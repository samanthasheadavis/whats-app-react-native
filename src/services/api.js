const mockMessages = [
  {
    incoming: true,
    message: "Hi Samantha"
  },
  {
    incoming: false,
    message: "Hey this is Samantha"
  },
  {
    incoming: true,
    message: "I know."
  }
]

export const getMockData = () => (
  new Promise(resolve => setTimeout(() => resolve(mockMessages),1000))
)
