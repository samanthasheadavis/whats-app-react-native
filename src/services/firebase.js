import * as firebase from "firebase";

export const initialize = () => firebase.initializeApp({
  apiKey: "AIzaSyD8B6kex_RG60jcaAXlajEqKNL0ldCIKrE",
  authDomain: "whats-app-clone-1d269.firebaseapp.com",
  databaseURL: "https://whats-app-clone-1d269.firebaseio.com",
  projectId: "whats-app-clone-1d269",
  storageBucket: "whats-app-clone-1d269.appspot.com",
  messagingSenderId: "295394625696"
})


export const setListener = (endpoint, updaterFn) => {
    firebase.database().ref(endpoint).on('value', updaterFn);
    return () => firebase.database().ref(endpoint).off();
}
