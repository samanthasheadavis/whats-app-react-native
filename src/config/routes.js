import Home from '../screens/Home';
import ChatScreen from '../screens/ChatScreen';
import CollapseLinear from '../screens/CollapseLinear'

 const routes = {
   home: { screen: Home },
   chat: { screen: ChatScreen },
   collapse: {screen: CollapseLinear}
 }

 export default routes;
