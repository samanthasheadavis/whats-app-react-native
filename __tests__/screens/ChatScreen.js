import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import routes from '../../src/config/routes';
import Home from '../../src/screens/Home';
import ChatScreen from '../../src/screens/ChatScreen';
import { getMockDataSpy } from '../../__mocks__/services/api';

jest.mock('../../__mocks__/services/api')

it('renders correctly', () => {
  const tree = renderer.create(
    <ChatScreen />
  );
  expect(tree).toMatchSnapshot();
});

it('has proper navigation options', () => {
  expect(ChatScreen.navigationOptions).toMatchSnapshot();
})

// it('navigate on press', () => {
//   const goBack = jest.fn();
//   const chatScreen = shallow(<ChatScreen navigation={{ navigate }} /> )
//   chatScreen.find('Button').simulate('press');
//   expect(goBack).toBeCalledWith("home");
// })

it('ChatScreen has proper navigation options', () => {
    expect(
        ChatScreen.navigationOptions({
            navigation: {
                state: {
                    params: {
                        name: 'Samantha'
                    }
                }
            }
        }
    )).toMatchSnapshot();
})

it('fetches data when component is rendered', () => {
    const chatScreen = renderer.create(<ChatScreen />);
    expect(getMockDataSpy).toBeCalled();
}) 
