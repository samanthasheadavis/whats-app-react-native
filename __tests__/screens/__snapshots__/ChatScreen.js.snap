// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen has proper navigation options 1`] = `
Object {
  "title": "Chat with Samantha",
}
`;

exports[`has proper navigation options 1`] = `[Function]`;

exports[`renders correctly 1`] = `
<RCTScrollView
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
